### DISCUSSION TITLE
"Creating EventLoop multiple times is not supported." after switching pages in single-page webapp

### DISCUSSION THREAD

**Discussion by SIGSTACKFAULT** [bevyengine/bevy#10373] [Q&A]
[nobody replied to my thread on the bevy discord](https://discord.com/channels/691052431525675048/1170179969847144478)

My bevy game is compiled to WASM and linked into a Vue app, which works fine *the first time you visit the page*.

But if you switch away from it (unmounting the vue component it's part of) and back it throws the error in the title.

[permalink to a version of my project which has this issue, for posterity](https://gitlab.com/SIGSTACKFAULT/fleetwright-web-designer/-/tree/c910f62a88668b1311963da581acfcda171d5967)

Here's a vjeo of me reproing it, or you can try it yourself on https://sigstackfault.gitlab.io/fleetwright-web-designer/#/vtt (observe it working, click on "editor" in the top-right, then on "vtt", then observe it not working)

https://github.com/bevyengine/bevy/assets/14184826/b4743c8c-c799-453a-9e10-3d6a47f62553

here's the Vue code that initializes WASM. It's inside a Vue component which is mounted by Vue Router:
```ts
wasm_init()
    .then(() => {
        renderer_loading.value = false;
        wasm_start();
    })
    .catch((e: Error) => {
        if (e.message.startsWith("Using exceptions for control flow")) {
            console.info("suppressing error: ", e.message);
        } else {
            console.error("caught error from wasm_init()");
            throw e;
        }
    });
```

I can run code when the component is unmounted but i couldn't figure out what

**Comment by naveenrobo** [comment]
@SIGSTACKFAULT I am getting the same error "Creating EventLoop multiple times is not supported.". Did you manage to solve this issue?

**Reply by naveenrobo** [reply]
It worked for me before updating to bevy 12. But in bevy 12 it is not rendering the second time. I had to refresh the page to get it loaded.
